mitodl/open-discussions
# switch course search to use grid design and course card

#### Pre-Flight checklist


#### What are the relevant tickets?

closes #2089

#### What's this PR do?

this switches over the course search UI to use a grid (instead of a vertical list of cards) and to use the new CourseCard component to render the courses.

I also tweaked the CourseCard slightly so that it now calls the callback to open the course drawer only when the course cover image or title are clicked, instead of when the whole card is clicked. I also ported over the behavior we had previously on the CompactCourseDisplay component, where clicking on one of the topics on the course would add that topic to the current selected search facets.

#### How should this be manually tested?

make sure that everything renders as it should on the course search. you should be able to search for things, scroll down on the infinite scroll, etc without any problems.

you should also be able to open the course detail drawer by clicking on the photo or the title for a course card, and clicking on one of the course's topics should add that topic to your search.

I think that's about it? the course carousel should also work as it did before, with no regressions.

#### Screenshots (if appropriate)

![computersearchfff](https://user-images.githubusercontent.com/6207644/61467910-71986e80-a94a-11e9-9432-d62c5c70f334.png)
![computersearch](https://user-images.githubusercontent.com/6207644/61467911-72310500-a94a-11e9-9c8e-78c1c8fdf815.png)
![computersearchmobbb](https://user-images.githubusercontent.com/6207644/61467959-86750200-a94a-11e9-91f5-55485ab5bd4d.png)

note that there isn't a mobile design yet, but the cards are at least functional.
